### PR TITLE
Add search for members on paginated member page (#369)

### PR DESCRIPTION
**Pull request by SeunghunSunmoonLee**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,8 @@ Next.js, Urql, Apollo, Prisma, PostgreSQL
 - Install dependencies in `/ui`
   - `yarn`
 - In /ui, copy `.env.local.default` to `.env.local`
+- In /ui, copy and paste DATABASE_URL string from `.env.development` to `.env.local`
```
Comment (Powersource) [inline README.md:17]: I don't think this should be needed? Both files are loaded.

But ah on checking `.env.local.default` we have `DATABASE_URL=""` in it which clears the variable. I think a better solution is to just remove that line.

File: README.md [hunk 2]
```diff
@@ -14,6 +14,8 @@ Next.js, Urql, Apollo, Prisma, PostgreSQL
...
+- put MAGIC_LINK_SECRET="fake" in `.env.local`
```
Comment (Powersource) [inline README.md:18]: It's easier if this is added to `.env.local.default` instead of the readme.

File: ui/components/EventMembers/index.tsx [hunk 3]
```diff
@@ -114,11 +118,30 @@ const EventMembers = ({ collection, currentUser }) => {
...
+
+  const items = useMemo(() => {
+    if (loading || !approvedMembers) {
+      return [];
+    }
+    return approvedMembers
+  }, [approvedMembers, loading]);
+
+  useEffect(() => {
+    // if (tooShortSearch) return;
```
Comment (Powersource) [inline ui/components/EventMembers/index.tsx:140]: don't we want this limitation?

File: ui/components/EventMembers/index.tsx [hunk 4]
```diff
@@ -114,11 +118,30 @@ const EventMembers = ({ collection, currentUser }) => {
       fetching: loading,
       error,
     },
+    searchApprovedMembers
   ] = useQuery({
     query: COLLECTION_MEMBERS_QUERY,
-    variables: { collectionId: collection.id, offset: 0, limit: 1000 },
+    variables: { collectionId: collection.id, usernameStartsWith: searchString, offset: 0, limit: 1000 },
```
Comment (Powersource) [inline ui/components/EventMembers/index.tsx:124]: @aerugo is it enough for now that we search by username or do we want e.g. name and/or email as well?

File: ui/components/SearchBar/index.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import React, { useEffect, useRef } from "react";
+import { SearchIcon, CloseIcon } from "../Icons";
+
```
Comment (Powersource) [inline ui/components/SearchBar/index.js:3]: We want to slowly move to ts for all files so better to have that by default when creating a new file. If you run into any annoying type issues it's fine to use `any` though

File: ui/components/SearchBar/index.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+export default ({ placeholder, collection, clearInput, value, onChange }) => {
+  const inputRef = useRef();
+  useEffect(() => {
+    inputRef.current.focus();
+  }, []);
```
Comment (Powersource) [inline ui/components/SearchBar/index.js:8]: i think we should use `autoFocus` on the element instead

**Review by Powersource — CHANGES_REQUESTED** [review]
Saw too late that it was still WIP so feel free to ignore the comments you were gonna adress anyway :grin:

**Comment by SeunghunSunmoonLee** [comment]
All suggestions are applied [d507619 ](https://github.com/cobudget/cobudget/commit/d5076190a4436f9063aad3cc9d71d9c8d403a14f)[now](https://github.com/cobudget/cobudget/pull/545#event-6051365199)

File: ui/components/EventMembers/index.tsx (hunk 4 above)
Comment (aerugo) [reply ui/components/EventMembers/index.tsx:123]: Lets do username for now, then add name and email later.

File: ui/components/EventMembers/index.tsx (hunk 3 above)
Comment (aerugo) [reply ui/components/EventMembers/index.tsx]: Yeah, I think one character makes sense when filtering a list where it's fine to have many results, but two characters are better for mentions - where you definitely never want to fill the autocomplete selector with all usernames starting with "a".

**Comment by aerugo** [comment]
@Powersource - you think we're ready to merge?

**Comment by Powersource** [comment]
merged this now, but with a few more commits. had to make the changes outside of this pr since i didn't have push access to this branch. feels like we should figure out a smoother collab setup for the next pr.

https://github.com/cobudget/cobudget/commit/80d6a47b6a1cad3ba4b7171b458a609bde355fe4
https://github.com/cobudget/cobudget/commit/628c27c677d1dd46ac64b56317bcb795657073a7
https://github.com/cobudget/cobudget/commit/de5d80f490eb9d0b90a4bbeb97746c140a0195c1

**Pull request merged**